app/controllers/users_controller.rb

```diff
@@ -54,11 +60,32 @@ def set_user
 
     # Only allow a list of trusted parameters through.
     def user_params
-      params.require(:user).permit([:display_name, :full_name, :family_name, :orcid, :email_messages_enabled])
+      @user_params ||= params.require(:user).permit([:display_name, :full_name, :family_name, :orcid, :email_messages_enabled, { collections_with_messaging: {} }])
```

Comment:
I'm not sure what role the `@user_params` plays?

Reply:
@mccalluc I'm guessing this is just a [memoizing ](https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/)

Reply:
Yes, sorry, the memoization is needed given that https://github.com/pulibrary/pdc_describe/pull/574/files#diff-cfdccd0a9d5df5a43aaad2a35d36ebbe187c52ad5fdc9846fa189d04537adb6eR77 invokes `#extract!`, updating the state of the HTTP request `params`. If this is not explicitly memoized, then `#extract!` does not mutate the `params` (please see https://api.rubyonrails.org/v6.1.6.1/classes/ActionController/Parameters.html#method-i-extract-21 for reference)